### ISSUE TITLE
Fix S3267 FN: Logical operators are not supported

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#7776]
Reported in https://github.com/SonarSource/sonar-dotnet/issues/7730

The original example from that issue contained both a False Positive (related to iterable non-enumerable objects, such as `ReadOnlySpan` and `Span`), and a False Negative, when a logical operator is used, instead of a logical pattern matching.

This issue  is about the False Negative part:

```cs
private static int GetWordCount(List<char> content)
{
    var wordCount = 0;
    foreach (var c in content)
    {
        if (c is ' ' or '\n' or '\r') // Noncompliant, TP
        if (c == ' ' || c == '\n' || c == '\r') // FN
        {
            wordCount++;
        }
    }

    return wordCount;
}
```

The FN has been extracted into a dedicated issue since it seems independent from the FP reported in https://github.com/SonarSource/sonar-dotnet/issues/7730.
Moreover, unlike the FP issue, this issue doesn't generate uncompilable code.
Repro added in https://github.com/SonarSource/sonar-dotnet/pull/7766.

**Comment by mary-georgiou-sonarsource** [comment]
This issue will be tracked in [NET-1255](https://sonarsource.atlassian.net/browse/NET-1255) (not publicly accessible)
